### DISCUSSION TITLE
`@wagmi/core` vs. `wagmi/actions`?

### DISCUSSION THREAD

**Discussion by firnprotocol** [wevm/wagmi#3334] [General]
these seem to be equivalent. which should i prefer? what's the difference?

more detail available on request. thanks in advance.

**Comment by 0xRowdy** [comment]
> these seem to be equivalent. which should i prefer? what's the difference?
> 
> more detail available on request. thanks in advance.

From my understanding, @wagmi/core is plain vanilla JavaScript and would be the go-to if you are not using React. With the package wagmi you will get all the react hooks out of the box but what if you need to use a plain function? 

For example, using react hook form and want to handle the prepareContractWrite and contractWrite functions inside the onSubmitHandler. Then you can't use the hooks but you could use the pure JS functions. This is where I would import from wagmi/actions to get access to those functions without having to install the @wagmi/core package.

Hope this helps.

**Reply by firnprotocol** [reply]
i already understand that part—my question is whether i have any need for `@wagmi/core` versus just importing from `wagmi/actions`. from your answer, it seems that the answer is "no".

a further question is how `@wagmi/core` compares to viem.

**Reply by 0xRowdy** [reply]
For that question, I'll point you to the tweet from the developers of viem & wagmi.

https://x.com/wevm_dev/status/1632814467083046912?s=20

**Reply by firnprotocol** [reply]
i am sorry man, but you keep telling me things i already know.

**Reply by 0xRowdy** [reply]
Then why are you asking...
